server/vulnerabilities/msrc/analyzer.go

```diff
@@ -180,7 +180,7 @@ func winBuildVersionGreaterOrEqual(feed, os string) (bool, error) {
 	for i := 0; i < 3; i++ {
 		if feedParts[i] != osParts[i] {
 			// comparing different product versions
-			return false, nil
+			return true, fmt.Errorf("different product versions: %s vs %s", feed, os)
```

Comment:
`return false, fmt.Errorf("...`?

Reply:
agree, i was struggling with this logic.  I think worth refactoring the function/function name.